drivers/sensor/icm20948/icm20948.h

```diff
@@ -0,0 +1,194 @@
+#define ICM20948_REG_EXT_SLV_SENS_DATA_18 0x4D
+#define ICM20948_REG_EXT_SLV_SENS_DATA_19 0x4E
+#define ICM20948_REG_EXT_SLV_SENS_DATA_20 0x4F
+#define ICM20948_REG_EXT_SLV_SENS_DATA_21 0x50
+#define ICM20948_REG_EXT_SLV_SENS_DATA_22 0x51
+#define ICM20948_REG_EXT_SLV_SENS_DATA_23 0x52
+
+/* User Bank 1 */
+#define ICM20948_REG_SELF_TEST_X_GYRO  0x02
+#define ICM20948_REG_SELF_TEST_Y_GYRO  0x03
+#define ICM20948_REG_SELF_TEST_Z_GYRO  0x04
+#define ICM20948_REG_SELF_TEST_X_ACCEL 0x0E
+#define ICM20948_REG_SELF_TEST_Y_ACCEL 0x0F
+#define ICM20948_REG_SELF_TEST_Z_ACCEL 0x10
+
+/* User Bank 2 */
+#define ICM20948_REG_GYRO_SMPLRT_DIV    0x00
+#define ICM20948_REG_GYRO_CONFIG_1      0x01
+#define ICM20948_REG_GYRO_CONFIG_2      0x02
+#define ICM20948_REG_ACCEL_SMPLRT_DIV_1 0x10
+#define ICM20948_REG_ACCEL_SMPLRT_DIV_2 0x11
+#define ICM20948_REG_ACCEL_CONFIG       0x14
+#define ICM20948_REG_ACCEL_CONFIG_2     0x15
+
+/* User Bank 3 */
+#define ICM20948_REG_I2C_MST_ODR_CONFIG 0x00
+#define ICM20948_REG_I2C_MST_CTRL       0x01
+#define ICM20948_REG_I2C_MST_DELAY_CTRL 0x02
+#define ICM20948_REG_I2C_SLV0_ADDR      0x03
+#define ICM20948_REG_I2C_SLV0_REG       0x04
+#define ICM20948_REG_I2C_SLV0_CTRL      0x05
+#define ICM20948_REG_I2C_SLV0_DO        0x06
+
+#define ICM20948_REG_I2C_SLV4_ADDR 0x13
+#define ICM20948_REG_I2C_SLV4_REG  0x14
+#define ICM20948_REG_I2C_SLV4_CTRL 0x15
+#define ICM20948_REG_I2C_SLV4_DO   0x16
+#define ICM20948_REG_I2C_SLV4_DI   0x17
+/* User banks common */
+#define ICM20948_REG_FIFO_EN_1     0x66
+
+#define ICM20948_REG_BANK_SEL 0x7F
+
+/* AK09916 */
+#define AK09916_I2C_ADDR 0x0C
+
+#define AK09916_WIA   0x01
+#define AK09916_ST1   0x10
+#define AK09916_HXL   0x11
+#define AK09916_HXH   0x12
+#define AK09916_HYL   0x13
+#define AK09916_HYH   0x14
+#define AK09916_HZL   0x15
+#define AK09916_HZH   0x16
+#define AK09916_ST2   0x18
+#define AK09916_CNTL2 0x31
+#define AK09916_CNTL3 0x32
+#define AK09916_TS1   0x33
+#define AK09916_TS2   0x34
+
+#define AK09916_DEVICE_ID 0x09
+
+#define ICM20948_ROOM_TEMP_OFFSET_DEG  21
+#define ICM20948_TEMP_SENSITIVITY_X100 33387
+#define ICM20948_SENS_READ_BUFF_LEN    22
+
+#define ICM20948_REG_I2C_SLV4_CTRL_VAL 0x80
+#define ICM20948_I2C_MST_STS_SLV4_DONE BIT(4)
+#define I2C_READ_FLAG                  BIT(7)
+
+#define ICM20948_MULT_MST_EN   BIT(7)
+#define ICM20948_I2C_MST_P_NSR BIT(4)
+#define ICM20948_I2C_MST_CLK   0x07
+
+struct icm20948_data {
+	int16_t accel_x;
+	int16_t accel_y;
+	int16_t accel_z;
+
+	int16_t temp;
+
+	int16_t gyro_x;
+	int16_t gyro_y;
+	int16_t gyro_z;
+
+	int16_t mag_x;
+	int16_t mag_y;
+	int16_t mag_z;
+	int16_t magn_scale_x;
+	int16_t magn_scale_y;
+	int16_t magn_scale_z;
+};
+
+struct icm20948_config {
+	struct i2c_dt_spec i2c;
+	uint8_t accel_fs;
+	uint8_t accel_hz;
+	uint8_t accel_fchoice;
+	uint8_t accel_lpf;
+	uint8_t gyro_fs;
+	uint8_t gyro_hz;
+	uint8_t gyro_fchoice;
+	uint8_t gyro_lpf;
+	uint8_t mag_freq;
+};
+
+enum accel_fs_sel {
```

Comment:
you might want to add icm20948_ prefix over here and likewise in the following enums

Reply:
done